runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -396,8 +396,18 @@ protected long getMaxGlobalFromMaxEntries(int maxEntries) {
 
         long size = streamAddressSpace.size();
         if (size == 0L) {
-            // The trim mark will allow to detect trimmed exceptions if the seeked address falls behind it
-            return streamAddressSpace.getTrimMark();
+            // We want to optimize and directly return when no updates are found in advance (i.e., no need to do remainingUpTo)
+            // but we need to consider the case when the 'syncUpTo' address falls below the trim mark, so we correctly
+            // report the TrimmedException allowing consumers to know that data is no longer available from this point.
+            // We could throw the TrimmedException directly or in this case let remainingUpTo discover it for us.
+            if (getCurrentGlobalPosition() <= streamAddressSpace.getTrimMark()) {
+                // If pointer is -1L we must return the known trim mark or the trimmed exception will be lost
+                return getCurrentGlobalPosition() == Address.NON_ADDRESS ? streamAddressSpace.getTrimMark() : getCurrentGlobalPosition();
```

Comment:
This -1 pass through logic is tied to how other functions set the GC trim mark and we should consider refactoring this code since it's very non-obvious.

Reply:
Not really, this -1 does not have to do with how other functions set the GC trim mark but instead is tied to what we discussed offline. 

We could directly throw a TrimmedException here (because we know for sure the position to which we want to sync up to falls behind the known trim mark). But, Maithem had suggested this method to limit its functions to return an address up to which to sync up to, and let remainingUpTo discover the trim exception. 

With that in mind, if the currentGlobalPosition is -1 (let's say the case where they want to sync the stream from the beginning of time), if we know that it falls behind the trim mark, we can't just return the current global position, otherwise syncing up to -1 is doing nothing, so we'll miss the known TrimmedException. So in this case we force it to sync up to the trim mark (which is above -1 and therefore the remainingUpTo will detect the TrimmedException).